qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -939,10 +940,27 @@ def _run_analysis(
         # pop arguments that are not given to fitter
         extra_options = self._arg_parse(**options)
 
-        # get experiment metadata
+        # get experiment metadata and run-options dependent data processor
         try:
             self.__experiment_metadata = experiment_data.metadata()
-        except AttributeError:
+
+            # No data processor has been provided at run-time we infer one from the job
+            # metadata and default to the data processor for averaged classified data.
+            if getattr(self, "__data_processor") is None:
+                try:
+                    run_options = self.__experiment_metadata["job_metadata"][-1]["run_options"]
+                except IndexError:
+                    run_options = {}
+
+                meas_level = run_options.get("meas_level", MeasLevel.CLASSIFIED)
+                meas_return = run_options.get("meas_return", "avg")
```

Comment:
These measurement settings are optional for most of experiments and assuming something in analysis base class is not right direction, i.e. at least you need to get default arguments from backend.run method. If some backend assumes different default arguments, this code doesn't work correctly.

Here I think we have two options

1. Set experiment base class run options.
https://github.com/Qiskit/qiskit-experiments/blob/112f4ce54865248c841973fc58564c975aac81df/qiskit_experiments/base_experiment.py#L271-L274
Most of experiments assumes classified, single data. Classified data cannot be averaged (see open qasm spec). If you forcibly set some defaults, you don't need to care about default arguments of backend run method.

2. Get these options from experiment data.
https://github.com/Qiskit/qiskit-experiments/blob/112f4ce54865248c841973fc58564c975aac81df/qiskit_experiments/database_service/db_experiment_data.py#L319-L324
We can copy this code add this to our `ExperimentData` until we switch to `DbExperimentDataV1`. Note that this is entry-wise information, thus we need to update processor logic, i.e. sort by options, then process entries from level0 to 2. This is reasonable because current implementation assumes all circuits have the same measurement settings, however, we can run experiment 1 with level1 and experiment 2 with level2, then merge two experiment data to perform analysis. This is edge case but still doable.

I think option2 is right direction in the long run, however, this is too much for the first release. Perhaps option1 is still ok for this PR.

Reply:
Agree that assuming default in the curve analysis is not the right place. See here: b143587.